sos/cleaner/__init__.py

```diff
@@ -643,7 +644,13 @@ def obfuscate_file(self, filename, short_name=None, arc_name=None):
             with open(filename, 'r') as fname:
                 for line in fname:
                     try:
-                        line, count = self.obfuscate_line(line)
+                        _parsers = [
+                            _p for _p in self.parsers if not
+                            any([
+                                re.match(p, short_name) for p in _p.skip_files
+                            ])
+                        ]
```

Comment:
Shouldnt we move these lines before `with open` cycle, to let feed static `_parsers` just once (per filename)? Or maybe yet "higher" as that move would still mean regenerating `_parsers` once per each file?

Reply:
Oh, yep - this should totally be outside of the `for line` loop.

Reply:
Regenerating once per file is the aim here - so that we have per-file selection of parsers.